src/open_dmlib.vala

```diff
@@ -729,6 +729,90 @@ namespace OpenDMLib
       return f;
     }
 
+    /**
+     * This class can read from network with a buffer
+     */
+    public class BufferedNetworkReader : BufferedReader
+    {
+      /**
+       * The connection which is used for read operations.
+       */
+      private SocketConnection conn;
+
+      /**
+       * This variable contains the number of read bytes.
+       */
+      private ssize_t bytes_read;
+
+      /**
+       * This constructor will create a new BufferedNetworkReader by connecting to
+       * a specified host and port.
+       * @param host The host to connect to
+       * @param port The port which should be used for connection
+       */
+      public BufferedNetworkReader( string host, uint16 port ) throws Error
+      {
+        SocketClient client = new SocketClient( );
+        this.conn = client.connect_to_host( host, port );
+        this.buffer = new uchar[ BUFFER_SIZE ];
+
+      }
+
+      /**
+       * @see BufferedReader.close
+       */
+      public override void close( )
+      {
+        this.conn.close( );
+        this.conn = null;
+      }
+
+      /**
+       * @see BufferedReader.next_data
+       */
+      public override size_t next_data( uchar[] data ) throws Error
+      {
+        ssize_t size = this.conn.input_stream.read( data );
+        this.bytes_read += size;
+
+        return size;
+      }
+
+      /**
+       * @see BufferedReader.skip
+       */
+      public override void skip( size_t skip_bytes )
+      {
+        if ( this.buffer_index + skip_bytes > this.buffer_bytes )
+        {
+          uint8[] tmp = new uint8[ skip_bytes - ( this.buffer_bytes - this.buffer_index ) ];
+          this.conn.input_stream.read( tmp );
```

Comment:
Unhandled error

Reply:
`this.bytes_read` is not considered and `this.conn.input_stream.read( tmp );` is not guaranteed to return tmp.length bytes